src/3d/qgsvectorlayerchunkloader_p.cpp

```diff
@@ -317,8 +317,13 @@ QVector<QgsRayCastingUtils::RayHit> QgsVectorLayerChunkedEntity::rayIntersection
         QVector3D nodeIntPoint;
         int triangleIndex = -1;
 
+        // the node geometry has been translated by chunkOrigin
+        // This translation is stored in the QTransform component
+        // this needs to be taken into account to get the whole transformation
+        const QMatrix4x4 nodeTransformMatrix = node->entity()->findChild<Qt3DCore::QTransform *>()->matrix();
+        const QMatrix4x4 fullTransformMatrix = nodeTransformMatrix * transformMatrix;
         // TODO: use also geo transform matrix here???
```

Comment:
geo transform matrix is now being applied, so this to-do can be removed

Reply:
Done.